homeassistant/components/rainmachine/switch.py

```diff
@@ -459,9 +463,18 @@ async def async_turn_off_when_active(self, **kwargs: Any) -> None:
     @raise_on_request_error
     async def async_turn_on_when_active(self, **kwargs: Any) -> None:
         """Turn the switch on when its associated activity is active."""
+        duration = kwargs.get(CONF_DURATION)
```

Comment:
Am I correct in thinking this logic is this?

1. If a duration has been configured, use it.
2. If not, and if (a) we want to use app run times and (b) the controller is providing runtimes, use them.

I would actually expect the reverse: if the user checks the "use app run times" box, that should be the default (falling back to the duration option if the controller isn't providing app run times). That seems like a cleaner way to understand how those two options interact.

Thoughts?

Reply:
Might be worth taking the breaking change and only using the times from the app so there are no two sources of truth

Reply:
You've got it backwards, but that may be the choice of naming. 

1. Use a duration parameter if provided (from a service call). 
2. Use app run time if enabled and we have one. 
3. Use CONF_ZONE_RUN_TIME, which is the default for all zones.

Reply:
My thinking is if you want to call a zone for a specific time from Home Assistant, you would still pass the duration as a service parameter. Otherwise just flipping the switch on a dashboard calls the zone with the app's time.

Just like setting a default on level for a switch, you'd want to be able to overrule it from a service call. 

If all else fails, use the default time for all zones (10min).

Reply:
@bachya @bdraco Commented code to clarify and revised string literal name for default global zone (did not change value so shouldn't affect any existing configs).

Reply:
I think your comments with the doc update in https://github.com/home-assistant/home-assistant.io/pull/24706 are good. 👍🏻